development/ts-migration-dashboard/app/components/App.tsx

```diff
@@ -35,8 +37,8 @@ export default function App() {
     string,
     BoxRect
   > | null>(null);
-  const boxesByModuleId =
-    boxRectsByModuleId === null
+  const boxesByModuleId = useMemo(() => {
```

Comment:
This was causing performance problems which were especially noticeable when adding the search feature.